src/swap/defi/rango.ts

```diff
@@ -356,18 +448,117 @@ export function makeRangoPlugin(opts: EdgeCorePluginOptions): EdgeSwapPlugin {
       throw new Error(`Rango could not fetch quote: ${responseText}`)
     }
 
-    const swap = asSwapResponse(await swapResponse.json())
+    const swapResponseJson = await swapResponse.json()
+    log(`Rango swap response:`, swapResponseJson)
+    const swap = asSwapResponse(swapResponseJson)
     const { route, tx } = swap
 
     if (swap.resultType !== 'OK') {
+      // Try to handle failed result
       if (
+        swap.resultType === 'NO_ROUTE' || // TODO: https://api-docs.rango.exchange/reference/getbestroutes - Integrate "diagnosisMessages" instead?
```

Comment:
Good enough for now, we'll just fail the binary search in the situation there truly is no route. Tabling this for later in case we run into our favorite bug where minimum violations are being misinterpreted as an unsupported swap.